### ISSUE TITLE
Customize does not support OpenStack provider

### ISSUE THREAD

**Issue by xmik** [test-kitchen/kitchen-vagrant#171]
A Hash of customizations to a Vagrant virtual machine supports some Vagrant providers, e.g. it does not support `openstack` provider. Real use case: when I set in .kitchen.yml:
```
    driver_config:
      box:	'debian-7.5-chef-11.16.4'
      ssh_key: '~/.ssh/id_rsa'
      customize:
        keypair_name: 'my-ssh-key'
```
The generated Vagrantfile should contain
```
  c.vm.provider :openstack do |p|
      p.keypair_name = "my-ssh-key"
  end
```
but instead it does:
```
  c.vm.provider :openstack do |p|
  end
```

**Comment by tyler-ball** [comment]
Hey @xmik - is there a reason you're using kitchen-vagrant instead of [kitchen-openstack](https://github.com/test-kitchen/kitchen-openstack)?  As you see in your PR, all of the logic for `customize` is driven through an ERB file which is hard to read, test and maintain.  If we can avoid adding more logic to it then we will.

**Comment by xmik** [comment]
@tyler-ball, thanks for response. I use kitchen-vagrant instead of kitchen-openstack:
* in order to avoid very long configuration in .kitchen.yml. Vagrant boxes make it possible. 
* kitchen-vagrant supports Windows, kitchen-openstack does not (as indicated in test-kitchen/kitchen-openstack#45, I also tried it and it did not work for me. However, there came a PR 4 days ago to support Windows, I didn't try it yet).
* I also use docker and wanted to avoid using 3 kitchen providers (kitchen-vagrant for OpenStack Windows, kitchen-openstack for OpenStack Linux machines, kitchen-docker for docker containers).

I saw the option that I can supply a custom Vagrantfile.erb, but I didn't want to supply it in each cookbook. 

it sounds reasonable to avoid catch-all `else` statement in Vagrantfile.erb. I'll change it to support only openstack vagrant provider and I'll add tests.
